app/controllers/api/v2/operatingsystems_controller.rb

```diff
@@ -7,7 +7,7 @@ class OperatingsystemsController < V2::BaseController
       end
 
       before_filter :find_optional_nested_object
-      before_filter :find_resource, :only => %w{show edit update destroy bootfiles}
+      before_filter :find_resource, :only => %w{edit update destroy bootfiles}
```

Comment:
the find method on Operatingsystem works differently from the rest of the models (it's find by id / name / description / fullname), when using find_resource show doesn't work like it's supposed to.